packages/arb-token-bridge-ui/src/components/App/App.tsx

```diff
@@ -390,46 +379,84 @@ function ConnectedChainSyncer() {
   return null
 }
 
+export function AppContentWrapper() {
+  const {
+    app: { connectionState }
+  } = useAppState()
+  const [tosAccepted] = useLocalStorage<string>(TOS_LOCALSTORAGE_KEY)
+  const isTosAccepted = tosAccepted !== undefined
+
+  if (!isTosAccepted) {
+    return null
+  }
+
+  if (connectionState === ConnectionState.NETWORK_ERROR) {
+    return (
+      <Alert type="red">
+        Error: unable to connect to network. Try again soon and contact{' '}
+        <a rel="noreferrer" target="_blank" href={GET_HELP_LINK}>
+          <u>support</u>
+        </a>{' '}
+        if problem persists.
+      </Alert>
+    )
+  }
+
+  return <AppContent />
+}
+
 export default function App() {
   const [overmind] = useState<Overmind<typeof config>>(createOvermind(config))
-  const [tosAccepted, setTosAccepted] =
+  const [isTosAccepted, setTosAccepted] =
```

Comment:
Previously, `isTosAccepted` was checking for `undefined` values only (`const isTosAccepted = tosAccepted !== undefined`), but this slightly changes the logic. Are we sure about this change?

Reply:
i think the `undefined` check is a legacy usage